Fixe(Stock entry from donation) solve receipt rendering problem

jeremielodi opened the pull request:
Closes #4751

jeremielodi committed:
Fixe(Stock entry from donation) solve receipt rendering issue

closes 4751

jniles commented on server/controllers/stock/reports/stock/entry_donation_receipt.js:
```diff
@@ -67,12 +67,13 @@ async function stockEntryDonationReceipt(documentUuid, session, options) {
     return row.total + aggregate;
   }, 0);
 
-  data.rows = rows;
```
Is there anyway to write a test for this to make sure it doesn't happen again?

mbayopanda reviewed (approved):
LGTM 👍

jeremielodi committed:
integration test

jeremielodi committed:
test before updates

jeremielodi committed:
using async

jniles commented on test/integration/stock/stock.js:
```diff
@@ -7,205 +7,174 @@ const shared = require('./shared');
-        helpers.api.created(res);
-      })
-      .catch(helpers.handler);
+  it('POST /stock/lots create a new stock lots entry', async () => {
```
async/await 💪

mbayopanda commented on test/integration/stock/stock.js:
```diff
@@ -7,205 +7,174 @@ const shared = require('./shared');
+  it('POST /stock/lots create a new stock lots entry', async () => {
+    const res = await agent.post('/stock/lots')
+      .send(shared.movementFirstLots);
+    helpers.api.created(res);
```
It seems like errors are not handled, i am wondering just if they can be handled

jeremielodi replied:
If an error occurs the status code won't be 201

```js
api.created = function created(res) {

  // make sure the response has correct HTTP headers
  expect(res).to.have.status(201);
  expect(res).to.be.json;

```

mbayopanda reviewed (approved):
👍 good job @jeremielodi 

I have made a comment about errors in e2e which are not handled, but since the test pass without error this comment is not a blocking one

jniles commented on test/integration/stock/stock.js:
```diff
@@ -7,205 +7,174 @@ const shared = require('./shared');
+    helpers.api.created(res1);
+    // get details of the movement
+    const docUuid = res1.body.uuid;
+    const res2 = await agent.get(`/stock/lots/movements?document_uuid=${docUuid}`);
```
This is really clean.  Great work @jeremielodi

jniles committed:
chore(stock): clean up donation tests slightly

Cleans up the donation tests to not fire up a PDF render just to assert
that the route isn't broken.

jniles reviewed (approved):
@jeremielodi This is great!

I've made a few changes to tighten up the tests.  Basically, I:

1. Changed the text "Stock Exit to Patient" to "Stock Exit to Donation"
2. Changed from using a PDF renderer (that would spin up chrome) to using the JSON renderer.  It should be much quicker.
3. Removed optional UUIDs.

This looks great to me now.
